### PR TITLE
Handle different treatment of date setting when new due < start

### PR DESCRIPTION
**Pull request by oliverguenther**
Unset start and duration if selecting a new finish date < start through the today link or datepicker.

https://community.openproject.org/wp/44191

### CODE REVIEW

**Comment by cbliard** [comment]
I found a bug in scenario 13 : If start date is not set, then it does not work as expected

image

I pushed a failing test case for it.

File: spec/features/work_packages/datepicker/datepicker_logic_spec.rb [hunk 1]
```diff
@@ -741,6 +741,142 @@ def apply_and_expect_saved(attributes)
...
+    let(:current_attributes) do
+      {
+        start_date: nil,
+        due_date: Date.parse('2021-02-18'),
+        duration: nil,
+        ignore_non_working_days: true
+      }
+    end
+
+    it 'sets finish date to selected date' do
```
Comment (cbliard) [inline spec/features/work_packages/datepicker/datepicker_logic_spec.rb]: You probably meant start date here.

**Comment by cbliard** [comment]
It made the feature test "reverses the dates if the selected date is before the current start date" fail, for obvious reason.
The special Scenario 13 is in contradiction with the failing test.

**Comment by oliverguenther** [comment]
> 

Correct, this spec is no longer valid with the new scenarios. I've removed it. Should be good to merge now

**Review by cbliard — APPROVED** [review]
Nice 👍️

**Pull request merged**